mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala

```diff
@@ -70,7 +70,8 @@ class SQLTransformer @Since("1.6.0") (@Since("1.6.0") override val uid: String)
     dataset.createOrReplaceTempView(tableName)
     val realStatement = $(statement).replace(tableIdentifier, tableName)
     val result = dataset.sparkSession.sql(realStatement)
-    dataset.sparkSession.catalog.dropTempView(tableName)
```

Comment:
It seems like a bug: when you cache a dataframe, create a view from the dataframe, and drop the view, Spark should not uncache the original dataframe. We can discuss more later.

Reply:
Yes. I also think it doesn't make a lot sense.

Reply:
Hmm, but the behavior is explicitly defined at `Catalog`. This should be a public API. We may not easily change it.